src/lib.rs

```diff
@@ -133,6 +133,12 @@ pub trait CustomizeConnection<C, E>: fmt::Debug + Send + Sync + 'static {
     fn on_acquire(&self, conn: &mut C) -> Result<(), E> {
         Ok(())
     }
+
+    /// Called with connections before they are dropped from the pool
+    ///
+    /// The default implementation does nothing
+    #[allow(unused_variables)]
+    fn on_release(&self, conn: &mut C) {}
```

Comment:
I don't know. While it does make sense semantically, that would allow the user to conserve the connection and I don't know if that's ok. Maybe `drop_conn()` should be one accepting the connection by value. I'll update the pull request.